### ISSUE TITLE
BigQuery truncate error

### ISSUE THREAD

**Issue by cdchan** [slingdata-io/sling-cli#98]
I'm getting this error when using the truncate mode with BigQuery

```
9:42PM INF execution failed
9:42PM ERR  error="googleapi: Error 400: Syntax error: Invalid empty identifier at [1:90], invalidQuery"

9:42PM INF Sling Replication Completed in 5m 341s | SOURCE_DB -> TARGET_DB

fatal:
~ 
~ failure running replication (see docs @ https://docs.slingdata.io/sling-cli)
~ Error executing create or replace table `***`.`source_table` as select * from `{schema}`.``***`.`source_table`` limit 0
googleapi: Error 400: Syntax error: Invalid empty identifier at [1:90], invalidQuery
```

The full-refresh mode works just fine with my config. For context, I just ran with the mode as full-refresh, then switched to truncate after setting the clustering column in BQ (https://cloud.google.com/bigquery/docs/creating-clustered-tables#modifying-cluster-spec). The docs suggest using the truncate mode to preserve DDL which is why I wanted to use truncate.

**Comment by flarco** [comment]
Hi can you run it in debug mode (flag -d) and share the log?

**Comment by cdchan** [comment]
I removed a bunch of memory and CPU stats in the middle

```
2024-01-03 22:16:46 INF importing into bigquery via local storage
2024-01-03 22:16:46 DBG writing to /tmp/bigquery/`source_schema`.`source_table_tmp`/2024-01-03T221646.682/part.01 [fileRowLimit=500000 fileBytesLimit=96000000 compression=GZIP concurrency=7 useBufferedStream=false fileFormat=csv]

...

2024-01-03 22:22:12 DBG select count(*) cnt from `source_schema`.`source_table_tmp`
2024-01-03 22:22:14 DBG create or replace table `source_schema`.`source_table ` as select * from `{schema}`.``source_schema`.`source_table `` limit 0
2024-01-03 22:22:14 DBG drop table if exists `source_schema`.`source_table_tmp`
2024-01-03 22:22:15 DBG table `source_schema`.`source_table_tmp` dropped
2024-01-03 22:22:15 INF execution failed
2024-01-03 22:22:15 ERR  error="~ execution failed\n--- task_run.go:85 func1 ---\n~ Could not WriteToDb\n--- task_run.go:635 runDbToDb ---\n~ Could not truncate table: `source_schema`.`source_table`\n--- task_run_write.go:434 WriteToDb ---\n~ Could not execute SQL\n--- database.go:1108 Exec ---\n~ Error executing create or replace table `source_schema`.`source_table` as select * from `{schema}`.``source_schema`.`source_table`` limit 0\n--- database_bigquery.go:250 ExecContext ---\ngoogleapi: Error 400: Syntax error: Invalid empty identifier at [1:90], invalidQuery"

2024-01-03 22:22:15 INF Sling Replication Completed in 5m 332s | SOURCE_DB -> TARGET_DB

fatal:
--- sling.go:421 func1 ---
--- sling.go:462 cliInit ---
--- cli.go:283 CliProcess ---
~ failure running replication (see docs @ https://docs.slingdata.io/sling-cli)
--- sling_logic.go:176 processRun ---
~ execution failed
--- task_run.go:85 func1 ---
~ Could not WriteToDb
--- task_run.go:635 runDbToDb ---
~ Could not truncate table: `source_schema`.`source_table`
--- task_run_write.go:434 WriteToDb ---
~ Could not execute SQL
--- database.go:1108 Exec ---
~ Error executing create or replace table `source_schema`.`source_table` as select * from `{schema}`.``source_schema`.`source_table`` limit 0
--- database_bigquery.go:250 ExecContext ---
googleapi: Error 400: Syntax error: Invalid empty identifier at [1:90], invalidQuery
```

**Comment by cdchan** [comment]
My replication YAML
```
source: SOURCE_DB
target: TARGET_DB

defaults:
  mode: full-refresh
  object: '{target_schema}.{stream_table}'

streams:
  source_schema.source_table:
    mode: truncate
    object: source_schema.source_table
```

**Comment by flarco** [comment]
Thanks, will release soon with fix. BigQuery did not support TRUNCATE so this was a work-around. Now it does support it (https://stackoverflow.com/a/64544540).

**Comment by flarco** [comment]
You can actually build it yourself if you want to try now: https://github.com/slingdata-io/sling-cli/pull/99

**Comment by flarco** [comment]
Closing, fixed in #99.

**Comment by flarco** [comment]
v1.0.68 Released.
